internal/services/elastic/elastic_tag_rule_resource.go

```diff
@@ -0,0 +1,297 @@
+package elastic
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elastic/legacysdk/elastic/mgmt/2020-07-01/elastic"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elastic/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elastic/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceElasticTagRule() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceElasticTagRuleCreateorUpdate,
+		Read:   resourceElasticTagRuleRead,
+		Update: resourceElasticTagRuleCreateorUpdate,
+		Delete: resourceElasticTagRuleDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ElasticTagRuleID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"monitor_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ElasticMonitorName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"rule_set_name": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  utils.String("default"),
+			},
+
+			"log_rules": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"send_aad_logs": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"send_subscription_logs": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"send_activity_logs": {
```

Comment:
Boolean attribute are always named as `something_enabled`

Reply:
Refer to [https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/logz/logz_tag_rule_resource.go](url)